src/Microsoft.Health.Dicom.Api/Features/Exceptions/ExceptionHandlingMiddleware.cs

```diff
@@ -90,6 +90,10 @@ private IActionResult MapExceptionToResult(Exception exception)
             case IOException io when io.Message.Equals("The request stream was aborted.", StringComparison.OrdinalIgnoreCase):
                 statusCode = HttpStatusCode.BadRequest;
                 break;
+            case DataStoreException ex when ex.IsExternal == true:
```

Comment:
Should we instead have a separate `Exception` like `ExternalStorageException` whose `InnerException` varies. That way we don't need to update our existing `Exception` types to accommodate. Then we just special-case this new exception here

Reply:
Although, I admit that by keeping the types we can preserve any logic that relies on `DataStoreException` and `ItemNotFoundException`. An alternative idea could be an interface that indicates the `Exception` may be from an external source like `IExternalException`

Reply:
Esma had the same suggestion. The code to throw these exceptions will get very verbose.

Reply:
At the very least, using an interface allows you to just have a case like:
```csharp
case IExternalException:
   // TODO
```

Instead of listing each implementation

Reply:
I have not seen interfaces used in Exceptions, only classes to allow for easy disambiguation, even though in this case we may not catch and throw based on the interface type.

Maybe we can reduce the need of specific exceptions like ItemNotFoundException. DataStoreException has the innerexception for details.
Only difference today is ItemNotFound is throwing 500 and DataStoreException is throwing  503. 

what do you think?

Reply:
Same idea as yours. Just used based class instead of interface

https://github.com/microsoft/dicom-server/pull/2605/commits/413c96606b583d7611436da624ce6fffe3598dc5

Reply:
Yeah, that makes sense. An alternative may just be to make `DataStoreException` the base exception, as we're really saying, "There was a problem with the data store." Optionally users can check the `IsExternal` property on the exception. Then `ItemNotFoundException` can derived from `DataStoreException`.

Reply:
Yeah maybe, right now DataStore as some additional properties that I don't want to touch.